packages/core/src/tsconfig.esnext.json

```diff
@@ -0,0 +1,12 @@
+{
+    "extends": "./tsconfig",
+    "compilerOptions": {
+        "target": "esnext",
+        "lib": [
```

Comment:
Would appreciate some thoughts on which libs should be included here! Seems like `es2017` should cover most of the features in newer browsers, but not sure if I should include any others.

Reply:
The lib doesn't change the output, just what you typecheck against.

Reply:
i would expect this to be the exact same as the other targets. given what @ericanderson said above, `lib` should not need to change between compiler targets. 

for instance, we still can't use `Promise` because we haven't declared it in [our environment requirements](https://blueprintjs.com/docs/#blueprint/getting-started.js-environment), regardless of compilation target.